weldx/util.py

```diff
@@ -23,34 +23,54 @@
     import weldx.transformations as tf
 
 
-def deprecated(func: Callable):
+def deprecated(since: str = None, removed: str = None, message: str = None) -> Callable:
     """Mark a functions as deprecated.
 
     This decorator emits a warning when the function is used.
 
+    Parameters
+    ----------
+    since :
+        The version that marked the function as deprecated
+    removed :
+        The version that will remove the function
+    message :
+        Additional information that should be added to the warning
+
+    Returns
+    -------
+    Callable :
+        Wrapped function
+
     Notes
     -----
     Original source: https://stackoverflow.com/a/30253848/6700329
 
-    Parameters
-    ----------
-    func :
-        The wrapped function
-
     """
 
-    @wraps(func)
-    def _new_func(*args, **kwargs):
-        warnings.simplefilter("always", DeprecationWarning)  # turn off filter
-        warnings.warn(
-            "Call to deprecated function {}.".format(func.__name__),
-            category=DeprecationWarning,
-            stacklevel=2,
-        )
-        warnings.simplefilter("default", DeprecationWarning)  # reset filter
-        return func(*args, **kwargs)
-
-    return _new_func
+    def _decorator(func):
+        @wraps(func)
+        def _new_func(*args, **kwargs):
+            wm = f"Call to deprecated function {func.__name__}.\n"
+            if since is not None:
+                wm += f"Deprecated since: {since}\n"
+            if removed is not None:
+                wm += f"Removed in: {removed}\n"
+            if message is not None:
+                wm += message
+
+            warnings.simplefilter("always", DeprecationWarning)  # turn off filter
+            warnings.warn(
+                wm,
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
+            warnings.simplefilter("default", DeprecationWarning)  # reset filter
```

Comment:
This should be the old state of the filter, not "default". This would create a hard to debug situation.

Reply:
Take a look at [this StackOverflow question](https://stackoverflow.com/questions/2536307/decorators-in-the-python-standard-lib-deprecated-specifically/30253848#30253848). Both of the highest-rated answers do it this way. Can't tell you why, but there must be a reason for it. I am not familiar with the warning system in python so I just copied the corresponding sections.

Reply:
I removed the filter settings and added our own deprecation warning type as discussed.